go/chat/signencrypt/codec_test.go

```diff
@@ -23,6 +23,24 @@ var plaintextInputs = []string{
 	strings.Repeat("long", 1000000),
 }
 
+type arbitraryMsg struct {
+	Message string `codec:"m" json:"m"`
+}
+type arbitraryNested struct {
+	Fun    arbitraryMsg `codec:"f" json:"f"`
+	Boring arbitraryMsg `codec:"b" json:"b"`
+}
+
+var associatedDataInputs = []interface{}{
+	"",
```

Comment:
I am curious what happens if you try to use nil as associated data. Probably the same as "", but it would be interesting to know and maybe disallow one of the two.

Reply:
it works just fine. as long as msgpack encodes it consistently, it really shouldn't matter what you pass in.

Reply:
added a test for it so the behavior is explicit.